roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,227 @@
+bids. Ideally on the node side we would already:
+
+1. Check signatures, remove invalid bids and disconnect peers providing such
+   bids.
+2. If a certain threshold of pending bids is reached, we can start dropping
+   lowest bids.
+
+TODO: How does this work with transactions? Can we even check signatures on the
+node side? If we don't check signatures on the node side, then how can we be
+spam resistant?
+
+Then the block producer should put bids into the block, limited by some maximum
+amount (don't overflow runtime memory). Higher bids take precedence obviously.
+
+If exposed how many parathread cores are available, that information can be
+taken into account by the block producer: E.g. if there are 3 cores available,
+10 blocks would mean approximately 30 bids are useful to have in the runtime
+right now. It would not be useful to put 10000 bids into a block for example.
+Block producers should then likely expire bids in the transaction pool already,
+if they don't make it after some to be specified lifetime.
+
+Expiring bids makes sense to keep resource usage down, but also helps with
+keeping expectations: If we say you bid now for a block asap, it would be out of
+expectations if your bid finally gets accepted two days later. This aspect would
+not need to be implemented on the node side though, but is already an
+optimization. We could also just import everything into the runtime and let the
+runtime drop what is too low (if resource limits allow for it).
+
+#### Runtime Implementation
+
+More details on auctioning
+[here](https://www.notion.so/paritytechnologies/Exotic-Scheduling-Auctions-53148e8d90d74411ab445b7d1dd6cfc8).
+
+- Bid has fee on top of bid for spam protection - fee can be quite low, it should
+  not harm the sender much, if losing a bid and a resend is necessary. Just high
+  enough so spam protection is maintained.
+- Minimum bid - configurable.
+- Lifetimes
+
+#### Possible Optimizations
+
+With the above, a user has to place an individual bid for each and every core
+desired. This is quite wasteful, if multiple cores are needed as a point in
+time. As each of these bids would likely be identical and each of them would
+need to be signed. What we can do is extend the format of the extrinsic slightly
+to contain a number specifying how many cores are desired and whether parallel
+cores are allowed (winning two cores in a single block). Likely a `max_parallel`
+parameter. Then the user can submit a single extrinsic, with only a single
+signature biding for e.g. 10 cores. Internally we treat those as indidivual
+equal bids (all the same price), competing individually with all the other bids.
+Therefore it is possible that only a fraction of the desired cores will actually
+be won. E.g. just one or two out of the ten desired.
+
+
+## Where is the money going?
+
+TODO: Bids and fees - where are they spent, what pallet do we need to interact here
+with?
```

Comment:
We should probably be interacting with the Balances pallet via the `ReservableCurrency` trait.